### PR TITLE
Add unit test for OpenAI provider

### PR DESCRIPTION
**Pull request by thiendsu2303**
## Background 

- Issue: #56 

This work is necessary to ensure the reliability and correctness of the OpenAI provider implementation in Ogem, particularly focusing on batch processing functionality and file handling. Proper test coverage helps catch potential issues early and maintains code quality.

## Changes 

- Fixed and enhanced test server implementation in [openai_test.go](cci:7://file:///Users/maichienvithien/YanoljaWorkspace/ogem/provider/openai/openai_test.go:0:0-0:0):
  - Added proper validation for multipart form data in file upload endpoint
  - Added method-specific handling (GET/POST) for batch and file endpoints
  - Fixed response format for file download endpoint
  - Improved error handling and validation in test server endpoints
- Added comprehensive test cases for:
  - File upload functionality
  - Batch job creation and status checking
  - File content download and response parsing
  - Error scenarios and edge cases

## Considerations 

- The test server now properly simulates the OpenAI API behavior for batch processing
- Test responses match the expected OpenAI API response format
- Error handling has been improved to provide better debugging information
- All tests are now passing successfully

## Remaining Tasks 

- Consider adding more edge cases and error scenarios
- Consider adding performance tests for batch processing
- Document the test patterns for future test additions

### CODE REVIEW

File: provider/openai/openai_test.go [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+			region:       "us",
+			baseUrl:      "://invalid-url",
+			apiKey:       "test-key",
+			wantErr:      true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			endpoint, err := NewEndpoint(tt.providerName, tt.region, tt.baseUrl, tt.apiKey)
```
Comment (nvtrinh2001) [inline provider/openai/openai_test.go:46]: It is recommended to write sub-tests with this format:
```
func TestNewEndpoint(t *testing.T) {
  t.Run()
  
  t.Run()
}
```

File: provider/openai/openai_test.go [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+			if tt.wantErr {
```
Comment (nvtrinh2001) [inline provider/openai/openai_test.go:47]: Conditional statements should be avoided
